### PR TITLE
chore: bump {gas-fee,network,selected-network,notification-services,profile-sync,signature}-controller

### PR DESCRIPTION
**Pull request by cryptodev-2s**
## **Description**

We want to ensure that mobile is using the latest versions of all controllers.

- Bump `@metamask/ppom-validator` from 0.35.1 to 0.36.0 ([view changes](https://github.com/MetaMask/ppom-validator/blob/main/CHANGELOG.md))
  - There are no notable changes between these versions.
- Bump `@metamask/gas-fee-controller` from 21.0.0 to 22.0.1 ([view changes](https://github.com/MetaMask/core/blob/main/packages/gas-fee-controller/CHANGELOG.md))
  - There are no notable changes between these versions.
- Bump `@metamask/signature-controller` from 22.0.0 to 23.1.0 ([view changes](https://github.com/MetaMask/core/blob/main/packages/signature-controller/CHANGELOG.md))
  - There are no notable changes between these versions.
- Bump `@metamask/profile-sync-controller` from 1.0.0 to 2.0.0 ([view changes](https://github.com/MetaMask/core/blob/main/packages/profile-sync-controller/CHANGELOG.md))
  - There are no notable changes between these versions.
- Bump `@metamask/selected-network-controller` from 18.0.2 to 19.0.0 ([view changes](https://github.com/MetaMask/core/blob/main/packages/selected-network-controller/CHANGELOG.md))
  - There are no notable changes between these versions.
- Bump `@metamask/network-controller` from 21.0.0 to 22.1.0 ([view changes](https://github.com/MetaMask/core/blob/main/packages/network-controller/CHANGELOG.md))
  - Notable changes in 22.1.0:
    - The `NetworkController:networkRemoved` messenger event will now be emitted when a network is removed.
    - New messenger actions `NetworkController:addNetwork`, `NetworkController:removeNetwork`, and `NetworkController:updateNetwork` which call the respective controller methods.
    - New `lastUpdatedAt`property to network configurations which will be set to the current time on addition or update
  - Notable changes in 22.0.2:
     - `getDefaultNetworkConfigurationsByChainId` returns the updated display names for `mainnet` and `linea`. `Ethereum Mainnet` instead of `Mainnet`, and `Linea` instead of `Linea Mainnet`.
- Bump `@metamask/notification-services-controller` from 0.11.0 to 0.14.0 ([view changes](https://github.com/MetaMask/core/blob/main/packages/notification-services-controller/CHANGELOG.md))
  - Notable changes in 0.12.0:
    - `deleteNotificationsById` now publishes `NotificationServicesController:notificationsListUpdated`
  - Notable changes in 0.12.1:
    - When notifications are disabled, list is now repopulated with Snap notifications

## **Related issues**

Completes https://github.com/MetaMask/MetaMask-planning/issues/3698 (and does more to satisfy peer dependency warnings).
https://github.com/MetaMask/metamask-mobile/issues/12383

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by Gudahtt** [comment]
Looks like there is another draft for this as well: #12565

**Comment by cryptodev-2s** [comment]
Closed as will be partially completed by #12565

**Comment by Gudahtt** [comment]
It looks like this PR only introduces two new peer dependency warnings:
```
warning " > @metamask/swaps-controller@11.0.0" has incorrect peer dependency "@metamask/gas-fee-controller@^21.0.0".
warning " > @metamask/swaps-controller@11.0.0" has incorrect peer dependency "@metamask/network-controller@^21.1.0".
```

I see that `@metamask/swaps-controller@12.0.0` would address those two warnings, but it involved a number of breaking changes that maybe would be better handled in a separate PR.

I can confirm that these peer dependency mismatches won't cause any problems. For the `gas-fee-controller` one, none of the types used by the `swaps-controller` were changed, and the `fetchGasFeeEstimates` function used has not changed. For `network-controller`, the `swaps-controller` was actually using a deprecated release with those breaking changes already present, so there is in fact no additional breaking change here.

**Review by tommasini — APPROVED** [review]

File: app/lib/ppom/ppom-util.test.ts [hunk 1]
```diff
@@ -157,6 +157,7 @@ describe('PPOM Utils', () => {
                 url: 'https://mainnet.infura.io/v3',
               },
             ],
+            lastUpdatedAt: Date.now(),
```
Comment (Gudahtt) [inline app/lib/ppom/ppom-util.test.ts:160]: I'm not able to understand the reason this was required. I see that leaving this out results in a type error though.

The change is harmless, so this is not a blocker. Must be something about how Jest's types work.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**